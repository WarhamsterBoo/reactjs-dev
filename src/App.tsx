import React from "react";
import { BrowserRouter, Link, Route, Switch } from "react-router-dom";
import { Header } from "./components/Header";
import { GameScreen } from "./screens/GameScreen";
import { LoginScreen } from "./screens/LoginScreen";
import { ForbiddenScreen } from "./screens/ForbiddenScreen";

export const App: React.FC<{}> = () => {
  return (
    <BrowserRouter>
      <ul>
        <li>
          <Link to="/login">Login</Link>
        </li>
        <li>
          <Link to="/forbidden">Forbidden</Link>
        </li>
        <li>
          <Link to="/">Game</Link>
        </li>
      </ul>
      <Switch>
        <Route exact path="/login">
          <LoginScreen />
        </Route>
        <Route exact path="/forbidden">
          <ForbiddenScreen />
        </Route>
        <Route exact path="/">
          <GameScreen />
        </Route>
      </Switch>
    </BrowserRouter>
  );
};
