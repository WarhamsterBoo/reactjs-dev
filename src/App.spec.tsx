import { shallow } from "enzyme";
import React from "react";
import { App } from "./App";

describe("App", () => {
  it("should render", () => {
    const sut = shallow(<App />);

    expect(sut).toMatchSnapshot();
  });
});
