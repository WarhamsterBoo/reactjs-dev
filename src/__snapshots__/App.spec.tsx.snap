// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render 1`] = `
<BrowserRouter>
  <ul>
    <li>
      <Link
        to="/login"
      >
        Login
      </Link>
    </li>
    <li>
      <Link
        to="/forbidden"
      >
        Forbidden
      </Link>
    </li>
    <li>
      <Link
        to="/"
      >
        Game
      </Link>
    </li>
  </ul>
  <Switch>
    <Route
      exact={true}
      path="/login"
    >
      <LoginScreen />
    </Route>
    <Route
      exact={true}
      path="/forbidden"
    >
      <ForbiddenScreen />
    </Route>
    <Route
      exact={true}
      path="/"
    >
      <Component />
    </Route>
  </Switch>
</BrowserRouter>
`;
